Entity+repository

Lim-Young-hun opened the pull request:
entitiy, repository , Mapper 추가 했습니다

Commit:
1. entity 추가

Commit:
1. dto 추가

Lim-Young-hun committed:
1. 엔티티 Mapper 연동되게 정리
2. Mapper 추가

Lim-Young-hun committed:
1. mapper Test중

Commit:
1. mapper, reposity 테스트 코드 추가

Lim-Young-hun commented on api/src/test/java/com/couponPayment/entitiy/mapper/MapperTest.java:
```diff
@@ -0,0 +1,258 @@
+                .ignoringFields("myWalletInfoId","userInfoId","storeInfoId" ,"walletReqId","requestDt","approvalDt","cancelDt")
+                .isEqualTo(transactionInfoTb);
+        assertThat(transactionInfoDto.getApprovalDt())
+                .isEqualTo("2025-03-10 09:23:27");
```
usingRecursiveComparison 돌면서 필드 값 dto <-> entity끼리 비교하는데
entity의 필드는 Timestamp인데, dto의 필드값은 String인 경우 어떻게 해결해야 하나요?

f-lab-lyan replied:
한쪽으로 맞추면 되지 않을까요? entity를 String으로 변환하거나 혹은 String을 다시 DT로 변환해볼 수 있겠죠. :-)

Lim-Young-hun replied:
지금 보니 한쪽으로 맞춰도 크게 상관이 없었네요...ㅎㅎ 감사합니다

Commit:
1. entity 정리

f-lab-lyan commented on api/src/main/java/com/couponPayment/entity/MyWalletInfoTb.java:
```diff
@@ -0,0 +1,55 @@
+@Getter
+@AllArgsConstructor
+@Entity
+public class MyWalletInfoTb extends BaseEntity{
```
그냥 취향의 문제이긴 합니다만, Entity를 만들었는데, 클래스 이름에 Tb가 들어갈 필요는 없지 않을까요? :-)

Lim-Young-hun replied:
그러면 보통 Entity 클래스 이름은 어떻게 정의 하시나요??

f-lab-lyan replied:
저희는 보통은 그냥 Table이름과 같게 하는 편입니다. :-)

f-lab-lyan commented on api/src/main/java/com/couponPayment/entity/mapper/UserInfoMapperHelper.java:
이 헬퍼 클래스들의 역할은 무엇인가요?

Lim-Young-hun replied:
ex) TransactionInfoDto to Entity로 맵핑 후 저장할때, 연관 관계로 잡혀있는 테이블의 정보도 같이 저장 하는 역할입니다.
class TransactionInfoDto{
  //외래키 
  Long useInfoId;
}
만약 헬퍼 클래스가 없다면, 연관 관계 테이블에 다른 칼럼값들은 null이 들어가고 id만 채워지게 됩니다

f-lab-lyan replied:
이 부분은 멘토링 시간에 다시 얘기해보도록 하죠. :-)

f-lab-lyan commented on api/src/test/java/com/couponPayment/entitiy/mapper/MapperTest.java:
```diff
@@ -0,0 +1,258 @@
+public class MapperTest {
+    /**
+    * 1. Mapper Test 목적
+    * 2. 각 Mapper를 통하여 save 후 find하여 각 Entity <-> Dto 맵핑 되는지 테스트
```
이런 목적이라면, Entity를 하나 만들고, 그걸 DTO로바꿔보고, 값이 같은지만 비교해봐도 되지 않을까요? Repository를 넣은 이유가 있는지 궁금합니다. :-)

Lim-Young-hun replied:
네 목적 자체에 좀 어긋난거 같아서 Mapper만 되는지 수정하겠습니다.

f-lab-lyan commented on api/src/test/java/com/couponPayment/repository/MyWalletInfoRepositoryTest.java:
```diff
@@ -0,0 +1,73 @@
+        when(myWalletInfoMapper.toEntity(myWalletInfoDto)).thenReturn(myWalletInfoTb);
+        when(myWalletInfoMapper.toDto(myWalletInfoTb)).thenReturn(myWalletInfoDto);
+
+        myWalletInfoTb = myWalletInfoRepository.save(myWalletInfoMapper.toEntity(myWalletInfoDto));
```
테스트하고 싶은 내용이 너무 많은 것 같아요. 디비에 저장이 잘 되는 것을 테스트하고 싶은 것인지, mapper가 잘 동작하는지 테스트하고 싶은지 명확히 하고 테스트를 분리하면 좋을 것 같습니다.

f-lab-lyan commented on api/src/test/java/com/couponPayment/repository/CommonReposity.java:
```diff
@@ -0,0 +1,10 @@
+
+@DataJpaTest
+@ExtendWith(MockitoExtension.class)
+public class CommonReposity {
```
개인적으로 이런 방식보다 어노테이션을 각각의 테스트에 붙이는 것이 더 좋다고 생각합니다. 이렇게 하면, ComonRepository에 코드가 들어가는 순간, 모든 테스트에 적용될 수도 있고, 각각의 테스트에서 보면, 어떤 어노테이션이 적용되었는지 명확하지 않기 때문입니다.

Lim-Young-hun replied:
넵 각각 붙이도록 하겠습니다.

f-lab-lyan commented on api/src/test/java/com/couponPayment/repository/MyWalletInfoRepositoryTest.java:
```diff
@@ -0,0 +1,73 @@
+    private MyWalletInfoRepository myWalletInfoRepository;
+
+    @MockitoBean
+    private MyWalletInfoMapper myWalletInfoMapper;
```
Mapper야 말로, Mappers.getMapper(...)를 통해서 가져올 수 있는 것이 아닌가요?

f-lab-lyan replied:
일반적으로 Mocking을 한단 얘기는 그 동작을 내가 제어하는 것이 목적인데, 여기서는 정확한 아웃풋만 내어주고 있는 것 같네요. :-)

Lim-Young-hun replied:
Mocking의 대한 이해도가 부족한거 같아서 수업시간 때 제가 다시 물어보고 수정하겠습니다

f-lab-lyan reviewed (commented):
MapStruct는 써보신 것인가요? 그렇다면, generated된 파일이 어떻게 생겼는지 얘기해보면 좋을 것 같습니다. :-) 그리고 전체적으로 formatter를 한번 적용시켜주면 좋을 것 같네요.

Commit:
1. mapper 정리 중

f-lab-lyan commented on api/src/test/java/com/couponPayment/repository/WalletReqRepositoryTest.java:
```diff
@@ -1,58 +1,56 @@
-                .usingRecursiveComparison()
-                .isEqualTo(walletReqTb);
+    void resetAutoIncrement() {
+        testEntityManager.getEntityManager()
```
이렇게 하는 것보다, 그냥 아이디가 save전에는 없는데, save이후에는 생겼다 ... 정도여도 괜찮지 않았을까요? :-)

f-lab-lyan commented on api/src/main/java/com/couponPayment/entity/dto/MyWalletInfoDto.java:
```diff
@@ -0,0 +1,27 @@
+    private String number;
+    private String cardType;
+    private String ownerType;
+
```
```suggestion
```

f-lab-lyan commented on api/src/main/java/com/couponPayment/entity/dto/TransactionInfoDto.java:
```diff
@@ -0,0 +1,41 @@
+@ToString
+public class TransactionInfoDto {
+
+    //private String cardId;
+    //private String orderNum;
+   // private String orderId;
+    //private Integer amount;
```
```suggestion
```

f-lab-lyan reviewed (approved):
수고하셨습니다. :-) Conflicts해결하고, 머지하도록 하죠!

Lim-Young-hun committed:
Toss api연동 및 api키 yml파일 암호화 진행 (#7)

* 1. 토스 빌링키 요청 서비스 추가
2. 환경변수 secretKey 추가 및 jasypt lib통해 yml api키 암호화
3. 테스트 코드(@SpringBootTest)에서 yml파일 읽을때 환경 변수는 읽지 않아서 따로 설정 해줘야함

* 1. 토스 빌링키를 통한 결제 요청 서비스 추가 중

* 1. yml api키 .env파일로 변경
2. 토스 결제 API 요청 테스트코드 추가

* 1. toss 결제 승인, 취소, 발급 테스코드 추가 및 정리

* 1. api, restemplate 추가

---------

Co-authored-by: younghun.lim <younghun.lim@seeroo.co.kr>

Lim-Young-hun committed:
Merge branch 'tossAPI연동' into entity+repository
